### ISSUE TITLE
LuaRocks doesn't clone submodules

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#2609]
LuaRocks invokes `git clone` without --recusive and then strips ".git" directory:

```
tarantoolctl rocks install mysql
Installing http://rocks.tarantool.org/mysql-scm-1.rockspec
Cloning into 'mysql'...
remote: Counting objects: 31, done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 31 (delta 1), reused 14 (delta 0), pack-reused 0
Receiving objects: 100% (31/31), 15.07 KiB | 0 bytes/s, done.
Resolving deltas: 100% (1/1), done.
-- The C compiler identification is GNU 7.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found TARANTOOL: /usr/include (found version "1.7.4-294-g54d06d7bf")
-- Tarantool LUADIR is /home/roman/tarantool/1.7/third_party/luarocks/.rocks/share/tarantool/rocks/mysql/scm-1/lua
-- Tarantool LIBDIR is /home/roman/tarantool/1.7/third_party/luarocks/.rocks/share/tarantool/rocks/mysql/scm-1/lib
CMake Error at CMakeLists.txt:37 (add_subdirectory):
  The source directory

    /tmp/luarocks_mysql-scm-1-bd2TXC/mysql/mariadb-connector-c

  does not contain a CMakeLists.txt file.


-- Configuring incomplete, errors occurred!
See also "/tmp/luarocks_mysql-scm-1-bd2TXC/mysql/build.luarocks/CMakeFiles/CMakeOutput.log".

Error: Build error: Failed cmake.
```

=> always clone repositories with  "--recursive".

**Comment by rtsisyk** [comment]
We can backport the fix from luarocks 3.0:  https://github.com/luarocks/luarocks/commit/e620a7e022c019ece6419446e6220d424aff9b42

**Comment by rtsisyk** [comment]
1.7.4-435-g0905310e1
